### PR TITLE
Fix S1144: Nested type constructor accessibility is wrong in the rule message

### LINKED ISSUE #7774: Fix S1144: Nested type constructor accessibility is wrong in the rule message
### Description

The message I got in VS IDE is:
image

This is not private ctor, but public.
And also this private class is used.

### Repro steps

This is the code example

```csharp
public class TagDto
{
    public int Id { get; set; }
    public string? Name { get; set; }
    public bool System { get; set; }
    public bool Active { get; set; }
    public DateTime? SynchronizedOn { get; set; }

    private class Mapping : Profile
    {
        public Mapping()
        {
            CreateMap<Tag, TagDto>();
        }
    }
}
```

And this is used as

`.ProjectTo<TagDto>(_mapper.ConfigurationProvider),`

### Related information

* C#/VB.NET Plugins version: SonarAnalyzer.CSharp Version 9.7.0.75501
* Visual Studio version: Version 17.7.0 Preview 5.0
* MSBuild / dotnet version: .NET 8 Preview 5
* Operating System: Windows

### PR DESCRIPTION
**Pull request by CristianAmbrosini**
Fixes #7774

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/Rules/UnusedPrivateMemberTest.Constructors.cs [hunk 1]
```diff
@@ -141,6 +141,25 @@ public interface IInterface
...
+{
+    private class Foo // Noncompliant
+    {
+        {{{accessModifier}}} Foo() // Noncompliant {{{{{expectedMessage}}}}}
+        {
+            var a = 1;
+        }
+    }
+}
+""").Verify();
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.Test/Rules/UnusedPrivateMemberTest.Constructors.cs:161]: The indentation can be improved.
```suggestion
        public void UnusedPrivateMember_NonPrivateConstructorInPrivateClass(string accessModifier, string expectedMessage) =>
            builder.AddSnippet($$$"""
                public class Some
                {
                    private class Foo // Noncompliant
                    {
                        {{{accessModifier}}} Foo() // Noncompliant {{{{{expectedMessage}}}}}
                        {
                            var a = 1;
                        }
                    }
                }
                """).Verify();
```

**Review by Tim-Pohlmann — APPROVED** [review]
Nicely done!

**Comment by Tim-Pohlmann** [comment]
Ah... one of my comments did not make it from my brain to the keyboard:
Add a UT with an `internal` class as well.

File: analyzers/tests/SonarAnalyzer.Test/Rules/UnusedPrivateMemberTest.Constructors.cs (hunk 1 above)
Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.Test/Rules/UnusedPrivateMemberTest.Constructors.cs:161]: There are three options:
1. Leave as is
2. Refactor everything
3. Clean as you code

I'd go with 3, but if you think it's too much of a sore on the eye, sticking with 1 is also fine.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/UnusedPrivateMember.cs [hunk 2]
```diff
@@ -410,6 +410,15 @@ public void M39()
     }
 }
 
+internal class MyClass
+{
+    protected MyClass() // Compliant
+    {
+        var a = 1;
+    }
+}
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/UnusedPrivateMember.cs:419]: Good test! I actually meant one with a private constructor (should be Noncompliant with a slightly different message).

Comment (CristianAmbrosini) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/UnusedPrivateMember.cs:419]: The message will be the same (the default) as the first DataRow in `UnusedPrivateMember_NonPrivateConstructorInPrivateClass`: `Remove the unused private constructor 'blabla'.`
I'll add this as well anyway 👍

Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/UnusedPrivateMember.cs:419]: You are absolutely right, of course. Still a good case to test!

**Pull request merged**